### ISSUE TITLE
Failure on cygwin

### ISSUE THREAD

**Issue by plicease** [plicease/Class-Inspector#5]
I was seeing errors like this on cpantesters coming from recent cygwin perl builds:

```
t/00_diag.t ............ ok
t/01_compile.t ......... ok

#   Failed test '->loaded_filename works'
#   at t/02_main.t line 58.

#   Failed test '->resolved_filename works'
#   at t/02_main.t line 60.
# Looks like you failed 2 tests of 54.
t/02_main.t ............ 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/54 subtests 
t/03_inc_to_local.t .... ok

#   Failed test 'loaded_filename works'
#   at t/04_main_functions.t line 60.

#   Failed test 'resolved_filename works'
#   at t/04_main_functions.t line 63.
# Looks like you failed 2 tests of 22.
t/04_main_functions.t .. 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/22 subtests 
```

https://gist.github.com/eeb2fbc87efa14d17b784e60d5b507a9

**Comment by plicease** [comment]
I was not able to reproduce this with the "system" perl on Cygwin.  I was able to reproduce this with by building a Perl 5.26.0 from source.  Note that there were some test failures in the perl build.  This seems to be a bug in Perl, and I was able to reproduce this bug in two lines of Perl:

https://gist.github.com/6ef280414576b3423608372106cef67a

```
xian-x86_64% perl badperl.pl 
bad perl at badperl.pl line 2.
```

The script should die on a bad perl and exit cleanly without output on a good perl.

**Comment by plicease** [comment]
Adding detection logic in `Makefile.PL` for bad perls to avoid them showing up in the cpantester matrix:

51b5f5e2c0c9b5ce82827d2ccc023bfe63138c12

**Comment by plicease** [comment]
This is apparently also a problem on x86_64 linux musl

**Comment by haarg** [comment]
Was this ever reported to p5p?

**Comment by plicease** [comment]
@haarg I did not personally report this to p5p, but I believe there was some discussion either on IRC or on the mailing list regarding this bug (more likely IRC because I don't frequently follow the mailing list).  I don't have the links handy.

**Comment by patrickbkr** [comment]
This seems still to be broken on 5.26.2. I just quickly searched RT and didn't find a corresponding bug. @plicease Could you, as someone a little knowledgeable about this bug, report a bug for this?

**Comment by plicease** [comment]
@patzim It is really easy to reproduce in affected perls using this two line Perl script:

https://gist.github.com/plicease/6ef280414576b3423608372106cef67a

In order to produce that error I had to build Perl from source under cygwin, which at the time that I tried it doesn't even pass the test suite.  Why someone was submitting cpantester results if perl itself wasn't even passing its own test suite I know not.  If cygwin is to be a supported platform I think the place to start is to get it to pass the test suite there.  I am ramping down my usage of cygwin as it has a lot of problems in my opinion, and there are better alternatives.  I do not have a good environment to test this anymore.  The bug was relevant at one point with musl as well, but I don't use that at all.  Happy to answer questions about this, but I don't really want to submit a bug report on platforms that I do not or will not be using on-going.
